File: src/main/java/ca/yyx/hu/aap/protocol/messages/VideoFocusEvent.java
package ca.yyx.hu.aap.protocol.messages;

import com.google.protobuf.nano.MessageNano;

import ca.yyx.hu.aap.AapMessage;
import ca.yyx.hu.aap.protocol.Channel;
import ca.yyx.hu.aap.protocol.MsgType;
import ca.yyx.hu.aap.protocol.nano.Protocol;

/**
 * @author algavris
 * @date 13/02/2017.
 */

public class VideoFocusEvent extends AapMessage {

    private static MessageNano makeProto(boolean gain, boolean unsolicited)
    {
        Protocol.VideoFocusNotification videoFocus = new Protocol.VideoFocusNotification();
        videoFocus.mode = gain ? 1 : 2;
        videoFocus.unsolicited = unsolicited;

        return videoFocus;
    }

    public VideoFocusEvent(boolean gain, boolean unsolicited)
    {
        super(Channel.ID_VID, MsgType.Media.VIDEOFOCUSNOTIFICATION, makeProto(gain, unsolicited));
    }
}

File: src/main/java/ca/yyx/hu/aap/protocol/messages/VideoFocusEvent.kt
package ca.yyx.hu.aap.protocol.messages

import ca.yyx.hu.aap.AapMessage
import ca.yyx.hu.aap.protocol.Channel
import ca.yyx.hu.aap.protocol.MsgType
import ca.yyx.hu.aap.protocol.nano.Media
import com.google.protobuf.nano.MessageNano

/**
 * @author algavris
 * *
 * @date 13/02/2017.
 */

class VideoFocusEvent(gain: Boolean, unsolicited: Boolean)
    : AapMessage(Channel.ID_VID, Media.MSG_MEDIA_VIDEOFOCUSNOTIFICATION, VideoFocusEvent.makeProto(gain, unsolicited)) {

    companion object {
        private fun makeProto(gain: Boolean, unsolicited: Boolean): MessageNano {
            val videoFocus = Media.VideoFocusNotification()
            videoFocus.mode = if (gain) 1 else 2
            videoFocus.unsolicited = unsolicited

            return videoFocus
        }
    }
}
